include/ozo/binary_deserialization.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <ozo/error.h>
+#include <ozo/type_traits.h>
+#include <ozo/concept.h>
+#include <ozo/detail/endian.h>
+
+namespace ozo {
+
+inline error_code recv(oid_t, const char* bytes, std::size_t size, std::string& value) {
+    // TODO: client to server encoding conversion?
+    value.assign(bytes, bytes + size);
+    return error::ok;
+}
+
+template <typename T>
+inline Require<std::is_integral<T>::value, error_code> recv(oid_t, const char* bytes, std::size_t size, T& value) {
+    if (size != sizeof(value)) {
+        return error::integer_value_size_mismatch;
+    }
+
+    const T* p_be = reinterpret_cast<const T*>(bytes);
+    value = detail::convert_from_big_endian(*p_be);
+    return error::ok;
+}
+
+template <typename OidMap, typename T>
+inline error_code convert(oid_t oid, const char* bytes, std::size_t size, const OidMap& oid_map, T& value) {
```

Comment:
IMO it is better to use ```value``` type or it's concept instead of these three parameters here:
```c++
oid_t oid, const char* bytes, std::size_t size
```